tutor/src/components/task/breadcrumbs.cjsx

```diff
@@ -28,29 +28,42 @@ module.exports = React.createClass
     currentStep: currentStep
 
   componentWillMount: ->
-    listeners = @getMaxListeners()
-    # TaskStepStore listeners include:
-    #   One per step for the crumb status updates
-    #   Two additional listeners for step loading and completion
-    #     if there are placeholder steps.
-    #   One for step being viewed in the panel itself
-    #     this is the + 1 to the max listeners being returned
-    #
-    # Only update max listeners if it is greater than the default of 10
-    TaskStepStore.setMaxListeners(listeners + 1) if listeners? and (listeners + 1) > 10
+    crumbs = @getCrumbs()
 
     # if a recovery step needs to be loaded, don't update breadcrumbs
     TaskStore.on('task.beforeRecovery', @stopUpdate)
     # until the recovery step has been loaded
     TaskStore.on('task.afterRecovery', @update)
 
     @startListeningForProgress()
-    crumbs = @getCrumableCrumbs()
+
+    @updateListeners(crumbs)
     @setState {crumbs}
 
   componentDidMount: ->
     @calculateCrumbsWidth()
 
+  updateListeners: (crumbs) ->
+    listeners = @getMaxListeners(crumbs)
+    # TaskStepStore listeners include:
+    #   One per step for the crumb status updates
+    #   Two additional listeners for step loading and completion
+    #     if there are placeholder steps.
+    #   One for step being viewed in the panel itself
+    #     this is the + 1 to the max listeners being returned
+    #
+    # Only update max listeners if it is greater than the default of 10
+    TaskStepStore.setMaxListeners(listeners + 1) if listeners? and (listeners + 1) > 10
+
```

Comment:
Thanks for the comment 😁  I'm pretty sure i've seen warnings about that before as well